graylog2-server/src/test/java/org/graylog/plugins/views/search/elasticsearch/IndexLookupTest.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.views.search.elasticsearch;
+
+import org.graylog2.indexer.ranges.IndexRange;
+import org.graylog2.indexer.ranges.IndexRangeService;
+import org.graylog2.plugin.database.Persisted;
+import org.graylog2.plugin.indexer.searches.timeranges.InvalidRangeParametersException;
+import org.graylog2.plugin.indexer.searches.timeranges.RelativeRange;
+import org.graylog2.plugin.streams.Stream;
+import org.graylog2.streams.StreamService;
+import org.joda.time.DateTime;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptySet;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class IndexLookupTest {
+
+    private IndexRangeService indexRangeService;
+    private StreamService streamService;
+    private IndexLookup sut;
+
+    @BeforeEach
+    void setUp() {
+        indexRangeService = mock(IndexRangeService.class);
+        streamService = mock(StreamService.class);
+        sut = new IndexLookup(indexRangeService, streamService);
+    }
+
+    @Test
+    void findsIndicesBelongingToStreamsInTimeRange() {
+
+        Set<String> streamIds = mockStreams("s-1", "s-2");
+
+        List<IndexRange> indexRanges = mockSomeIndexRanges();
+
+        IndexRange matchingIndexRange = indexRanges.get(0);
+
+        sut.indexRangeContainsOneOfStreams = (i, s) -> i.equals(matchingIndexRange);
+
+        Set<String> result = sut.indexNamesForStreamsInTimeRange(streamIds, someTimeRange());
+
+        assertThat(result).containsExactly(matchingIndexRange.indexName());
+    }
+
+    private RelativeRange someTimeRange() {
```

Comment:
just a nitpick, but wouldn't it be more readable if the private function comes after the test?

Reply:
:+1: Yup, moved it down. I placed it after the first usage in the first test, but I agree that it should go with the other helpers to the bottom of the file.